tabular/src/autogluon/tabular/models/abstract/model_trial.py

```diff
@@ -24,33 +25,40 @@ def model_trial(args, reporter: LocalStatusReporter):
     try:
         model, args, util_args = prepare_inputs(args=args)
 
+        epochs = util_args.pop("epochs")
+
+        report_probs = util_args.pop('report_probs')
+
         X_train, y_train = load_pkl.load(util_args.directory + util_args.dataset_train_filename)
         X_val, y_val = load_pkl.load(util_args.directory + util_args.dataset_val_filename)
 
         fit_model_args = dict(X_train=X_train, y_train=y_train, X_val=X_val, y_val=y_val)
         predict_proba_args = dict(X=X_val)
-        model = fit_and_save_model(model=model, params=args, fit_args=fit_model_args, predict_proba_args=predict_proba_args, y_val=y_val,
-                                   time_start=util_args.time_start, time_limit=util_args.get('time_limit', None), reporter=None)
+        model = fit_and_save_model(model=model, params=args, fit_args=fit_model_args, predict_proba_args=predict_proba_args,
+                                   y_val=y_val, time_start=util_args.time_start, time_limit=util_args.get('time_limit', None),
+                                   epochs=epochs, reporter=reporter, report_probs=report_probs)
     except Exception as e:
-        if not isinstance(e, TimeLimitExceeded):
+        if not (isinstance(e, TimeLimitExceeded) or isinstance(e, AutoGluonEarlyStop)):
             logger.exception(e, exc_info=True)
+        if isinstance(e, AutoGluonEarlyStop):
+            model.save()
         reporter.terminate()
-    else:
-        reporter(epoch=1, validation_performance=model.val_score)
 
 
 def prepare_inputs(args):
     task_id = args.pop('task_id')
     util_args = args.pop('util_args')
 
+    args.pop("epochs")
+
     file_prefix = f"trial_{task_id}"  # append to all file names created during this trial. Do NOT change!
     model = util_args.model  # the model object must be passed into model_trial() here
     model.name = model.name + os.path.sep + file_prefix
     model.set_contexts(path_context=model.path_root + model.name + os.path.sep)
     return model, args, util_args
 
 
-def fit_and_save_model(model, params, fit_args, predict_proba_args, y_val, time_start, time_limit=None, reporter=None):
+def fit_and_save_model(model, params, fit_args, predict_proba_args, y_val, time_start, time_limit=None, epochs=1, reporter=None, report_probs=False):
```

Comment:
In general, I prefer default values =None, and deal with it in the function, so epochs=None, report_probs=None

Reply:
code updated